### PR TITLE
Replace hetero-list YUI button and menu with new style button and tippy.js menu

### PR DESCRIPTION
**Pull request by mawinter69**
same as #8340 which was reverted with #8412

### Proposed changelog entries

- Update several buttons and menus to replace YahooUI in more locations.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8418"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

File: war/src/main/js/components/dropdowns/hetero-list.js
Comment (dwnusbaum) [inline war/src/main/js/components/dropdowns/hetero-list.js:1]: Maybe you are already aware of this issue, but other plugins used the adjunct directly and the deletion of this file at least causes warnings (IDK if it breaks things or not). See [this search](https://github.com/search?type=code&q=+owner%3Ajenkinsci+%3Cst%3Aadjunct+includes%3D%22lib.form.hetero-list.hetero-list%22). You can reproduce the warnings by opening a page that uses the adjunct (e.g. a Multibranch Pipeline configuration page) while running the latest weekly. Here is what the warnings look like:

```
2023-09-26 16:19:11.029+0000 [id=116]	WARNING	o.k.s.f.adjunct.AdjunctsInPage#findNeeded: No such adjunct found: lib.form.hetero-list.hetero-list
org.kohsuke.stapler.framework.adjunct.NoSuchAdjunctException: Neither lib.form.hetero-list.hetero-list.css, .js, .html, nor .jelly were found
	at org.kohsuke.stapler.framework.adjunct.Adjunct.<init>(Adjunct.java:125)
	at org.kohsuke.stapler.framework.adjunct.AdjunctManager.get(AdjunctManager.java:148)
	at org.kohsuke.stapler.framework.adjunct.AdjunctsInPage.findNeeded(AdjunctsInPage.java:161)
	at org.kohsuke.stapler.framework.adjunct.AdjunctsInPage.generate(AdjunctsInPage.java:113)
	at org.kohsuke.stapler.jelly.AdjunctTag.doTag(AdjunctTag.java:82)
        ... more Jelly/Jetty/Stapler stack frames ...
```

**Comment by MarkEWaite** [comment]
Confirmed by a user that the multibranch Pipeline plugin no longer allows the addition of multiple branch sources after the first save.  Described in [JENKINS-72170](https://issues.jenkins.io/browse/JENKINS-72170) and confirmed by bisect that this commit introduced the issue.

I think that a fix for this will need to be included as a backport into Jenkins 2.426.1 LTS.  I'm trying some alternatives now.

**Comment by mawinter69** [comment]
taking a look already

**Comment by MarkEWaite** [comment]
In case it helps your investigation, I tried an experiment by adding a few files back to Jenkins core.  The experiment passed my interactive tests with multibranch Pipeline plugin.

https://github.com/MarkEWaite/jenkins/commit/a4ac5a9b3128b4e7a8c70f68d7b3ea62cec62af3 is the commit on my https://github.com/MarkEWaite/jenkins/commits/restore-hetero-list-jelly branch

**Comment by mawinter69** [comment]
Fixed with #8602

File: war/src/main/js/components/dropdowns/hetero-list.js
Comment (mawinter69) [reply war/src/main/js/components/dropdowns/hetero-list.js:1]: The adjunct declaration can be removed in the plugins once they require Jenkins 2.422 or newer.

Comment (jtnord) [reply war/src/main/js/components/dropdowns/hetero-list.js:1]: > The adjunct declaration can be removed in the plugins once they require Jenkins 2.422 or newer.

and unless that happens your logs now contain massive stack traces :(

**Pull request merged**
